homeassistant/helpers/template.py

```diff
@@ -2029,9 +2029,11 @@ def from_json(value):
     return json_loads(value)
 
 
-def to_json(value, ensure_ascii=True):
+def to_json(value, ensure_ascii=True, indent=None, sort_keys=False):
     """Convert an object to a JSON string."""
-    return json.dumps(value, ensure_ascii=ensure_ascii)
+    return json.dumps(
+        value, ensure_ascii=ensure_ascii, indent=indent, sort_keys=sort_keys
+    )
```

Comment:
Looking at it a bit, it looks like the helper version is made for serializing home assistant objects and handles things like custom types, date encodings, etc. I'm not sure that's the right fit for Jinja, but open to input.

Reply:
Even using orjson directly would be fine. The performance issues are with the built in json lib

Reply:
Got it. Will explore making the change.

Reply:
As far as I can tell, orjson doesn't support the `ensure_ascii` option or arbitrary indentation (anything other than 2 spaces) for pretty printing.  Switching would be a breaking change for existing `to_json` users.  I suppose we could support the legacy options by using the old library and limit the new options to what orjson supports, but it feels like switching libraries may be beyond the scope of this change.

Reply:
It looks like ensure_ascii is the default 🙈. While it makes sense for historical reasons that's not very friendly to users who's language isn't primarily represented by ascii

Reply:
FWIW, I think it's ok as a default since it just makes the serializer dump those strings using Unicode codepoints (e.g. `\uxxxx`), which are the most compatible option supported in the json spec. But if you prefer to migrate first I can close this PR and make a migration PR. It just won't be possible to cleanly migrate sans breaking change without introducing a differently-named filter. Also, features like preferring ascii and custom indentation/formatting won't be supportable with orjson. But we can get somewhat close.

Reply:
I took a look through the community forums to see how people are using this function and I didn't come up with anything help.

On one hand I'm not sure that its even used enough that performance is a concern, but I don't want to hamstring us later since its almost always worse to make a breaking change later.

Reply:
My biggest question would be whether the quirks of orjson are the ones we want to live with in the API -- Hyrum's law and all 😉. It seems at a glance like a fairly inflexible API (sacrificing some usability and flexibility for performance, which is often the right trade off as internal implementation detail but maybe less nice for developer experience) and would take things like uuids and datetimes without a straightforward way to turn those features off -- quirks we'd need to support in perpetuity.

Reply:
One way out of this would be to deprecate the existing `to_json` (leaving it in place but deprecated/not recommended) and replace it with `as_json` using the new library.  The existing conversion APIs like `as_datetime`, `as_local` use the `as` naming scheme instead of the `to` naming scheme, so it would be more consistent anyway.  Deserializing may be possible compatibly already.

Reply:
> My biggest question would be whether the quirks of orjson are the ones we want to live with in the API -- Hyrum's law and all 😉.

90-95% of the code base has already moved to it so I think thats not a concern we need to retread.

> It seems at a glance like a fairly inflexible API (sacrificing some usability and flexibility for performance, which is often the right trade off as internal implementation detail but maybe less nice for developer experience) and would take things like uuids and datetimes without a straightforward way to turn those features off -- quirks we'd need to support in perpetuity.

I don't think templates are going to have `dataclass` or `UUID` objects in templates, but we already serialize `datetime` to `isoformat` almost everywhere else so it lines up nicely.

Reply:
> One way out of this would be to deprecate the existing `to_json` (leaving it in place but deprecated/not recommended) and replace it with `as_json` using the new library. The existing conversion APIs like `as_datetime`, `as_local` use the `as` naming scheme instead of the `to` naming scheme, so it would be more consistent anyway. Deserializing may be possible compatibly already.

That seems like a reasonable path forward 👍

Reply:
actually -- from_json is already using orjson.  So it's just to_json.  I think I can set a default handler and turn on passthroughs to disable all of the nonstandard types (at least at the outset so we wouldn't be stuck with them).  I'm going to give that a shot.  Will update this PR with the proposal.

Reply:
OK -- gave it a shot.  Please take a look.  I'm not sure if I did all the things right for deprecation here (and certainly we'd need to document the change).